java/memory/memory-core/src/main/java/org/apache/arrow/memory/BufferLedger.java

```diff
@@ -306,7 +311,7 @@ ArrowBuf newArrowBuf(final long length, final BufferManager manager) {
   @Override
   public ArrowBuf retain(final ArrowBuf srcBuffer, BufferAllocator target) {
 
-    if (BaseAllocator.DEBUG) {
+    if (BaseAllocator.DEBUG && historicalLog != null) {
```

Comment:
nit: one more BaseAllocator.DEBUG we can remove

Reply:
Thanks, changed.